DetectorDescription/OfflineDBLoader/src/DDCoreToDDXMLOutput.cc

```diff
@@ -784,25 +784,27 @@ void DDCoreToDDXMLOutput::position(const TGeoVolume& parent,
   xos << "<rChild name=\"" << childVolName << "\"/>" << std::endl;
 
   const auto matrix = child.GetMatrix();
-  if (matrix != nullptr && !matrix->IsIdentity()) {
```

Comment:
@iarspider, if (matrix != nullptr && !matrix->IsIdentity()) is a valid construction. If the first check @false@ the second does not checked.

Reply:
@civanch I have split the condition to put https://github.com/cms-sw/cmssw/pull/46075/files#diff-de5a97aa3932a8d881fe699bde286ceabe88e31003ce97291f0ac2a7dba8d5bdR801-R806 inside `if matrix` check

Reply:
A rotation matrix may or may not be present in a transformation matrix. Sometimes a child is placed in a mother with no translation and no rotation. In Geant4 one needs to provide a translation vector with a default 3-vector for the translation matrix while in DD4hep one may specify the placement dropping reference to both translation and rotation

Reply:
@bsunanda , should we review this code or accept this fix?